util/rlp/src/stream.rs

```diff
@@ -273,6 +317,17 @@ impl RlpStream {
 			false => panic!()
 		}
 	}
+
+	/// Finalize current ubnbound list. Panics if no unbounded list has been opened.
+	pub fn complete_unbounded_list(&mut self) {
+		let list = self.unfinished_lists.pop().expect("No open list.");
+		if list.max.is_some() {
+			panic!("List type mismatch.");
+		}
+		let len = self.buffer.len() - list.position;
+		self.encoder().insert_list_payload(len, list.position);
+		self.note_appended(1);
+	}
```

Comment:
There is a use case when both bounded and unbounded lists are added to the same stream. I'd redesign `RlpStream` to accept tuples instead of fixed size `begin_list()`/`end_list()` and add a variant of `append()` that accepts an iterator. But that's out of the scope of this PR